api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceServiceImpl.java

```diff
@@ -102,15 +120,55 @@ public void updateUserRoles(String ns, String id, Set<WorkspaceUserRole> userRol
         currentUserRole.setRole(mapValue.getRole());
         userRoleMap.remove(currentUserRole.getUser().getUserId());
       } else {
+        // This is how to remove a user from the FireCloud ACL:
+        // Pass along an update request with NO ACCESS as the given access level.
+        WorkspaceACLUpdate removedUser = new WorkspaceACLUpdate();
+        removedUser.setEmail(currentUserRole.getUser().getEmail());
+        removedUser.setCanCompute(false);
+        removedUser.setCanShare(false);
+        removedUser.setAccessLevel(WorkspaceAccessLevel.NO_ACCESS.toString());
+        updateACLRequestList.add(removedUser);
         dbUserRoles.remove();
       }
     }
 
     for (Map.Entry<Long, WorkspaceUserRole> remainingRole : userRoleMap.entrySet()) {
       dbWorkspace.getWorkspaceUserRoles().add(remainingRole.getValue());
     }
-    // TODO(calbach): This save() is not technically necessary but included to
-    // workaround RW-252. Remove either this, or @Transactional.
-    workspaceDao.save(dbWorkspace);
+
+    for(WorkspaceUserRole currentWorkspaceUser : dbWorkspace.getWorkspaceUserRoles()) {
+      WorkspaceACLUpdate currentUpdate = new WorkspaceACLUpdate();
+      currentUpdate.setEmail(currentWorkspaceUser.getUser().getEmail());
+      currentUpdate.setCanCompute(false);
+      if (currentWorkspaceUser.getRole() == WorkspaceAccessLevel.OWNER) {
+        currentUpdate.setCanShare(true);
+        currentUpdate.setAccessLevel(WorkspaceAccessLevel.OWNER.toString());
+      } else if (currentWorkspaceUser.getRole() == WorkspaceAccessLevel.WRITER) {
+        currentUpdate.setCanShare(false);
+        currentUpdate.setAccessLevel(WorkspaceAccessLevel.WRITER.toString());
+      } else {
+        currentUpdate.setCanShare(false);
+        currentUpdate.setAccessLevel(WorkspaceAccessLevel.READER.toString());
+      }
+      updateACLRequestList.add(currentUpdate);
+    }
+    try {
+      WorkspaceACLUpdateResponseList fireCloudResponse = fireCloudService.updateWorkspaceACL(ns, id, updateACLRequestList);
+      if (fireCloudResponse.getUsersNotFound().size() != 0) {
```

Comment:
Are there any other errors that could occur that we need to handle here?

Reply:
Not that won't simply throw an exception from FC.